docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,562 @@
+different sources such as databases, web services or file systems. These data
+sources are subject to latency, and we'd prefer to query them efficiently.
+
+If we are just reading data, we can make a series of optimizations such as:
+
+ - batching requests to the same data source
+ - requesting independent data from different sources in parallel
+ - caching previously seen results
+
+However, if we mix these optimizations with the code that fetches the data
+we may end up trading clarity for performance. Furthermore, we are
+mixing low-level (optimization) and high-level (business logic with the data
+we read) concerns.
+
+# Installation
+
+First of all, add the following dependency to your SBT build file:
+
+```scala
+"com.fortysevendeg" %%% "fetch" % "0.1.0"
+```
+
+Now you'll have Fetch available both in Scala and Scala.js.
+
+# Usage
+
+In order to tell Fetch how to retrieve data, we must implement the `DataSource` typeclass.
+
+```scala
+trait DataSource[Identity, Result]{
+  def fetch(ids: List[Identity]): Eval[Map[Identity, Result]]
+}
+```
+
+It takes two type parameters:
+
+ - `Identity`: the identity we want to fetch (a `UserId` if we were fetching users)
+ - `Result`: the type of the data we retrieve (a `User` if we were fetching users)
+
+The `fetch` method takes a non empty list of identities and must return an `Eval` that will result
+in a map from identities to results. Accepting a list of identities gives Fetch the ability to batch requests to
+the same data source, and returning a mapping from identities to results Fetch can detect whenever an identity
+couldn't be fetched or no longer exists.
+
+Returning `Eval` makes it possible to defer evaluation with a concurrency monad when running a fetch.
+
+## Writing your first data source
+
+Now that we know about the `DataSource` typeclass, let's write our first data source! We'll start implementing a data
+source for fetching users given their id. The identity type will be a user's id and the result type a user.
+
+```scala
+import cats.Eval
+
+import fetch._
+
+type UserId = Int
+case class User(id: UserId, username: String)
+
+implicit object UserSource extends DataSource[UserId, User]{
+  override def fetch(ids: List[UserId]): Eval[Map[UserId, User]] = {
+    Eval.later({
+      println(s"Fetching users $ids")
+	  ids.map(i => (i, User(i, s"@egg_$i"))).toMap
+    })
+  }
+}
+```
+
+Now that we have a data source we can write a function for fetching users
+given an id, we just have to pass a `UserId` as an argument to `Fetch`.
+
+```scala
+def getUser(id: UserId): Fetch[User] = Fetch(id) // or, more explicitly: Fetch(id)(UserSource)
+```
+
+## Declaring and running a fetch
+
+We are now ready to declare and run fetches. We need to provide Fetch with a target
+monad when we want to execute a fetch. We'll be using `Id` for now, make sure to import
+`fetch.implicits._` since Fetch needs an instance of `MonadError[Id, Throwable]` for running
```

Comment:
Also for any other Monads. We should state right here that Fetch comes with builtin impls of MonadError for Id, Task, Future, etc...